gateway/modules/crud/bolt/create.go

```diff
@@ -0,0 +1,100 @@
+package bolt
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/boltdb/bolt"
+	"github.com/sirupsen/logrus"
+
+	"github.com/spaceuptech/space-cloud/gateway/model"
+	"github.com/spaceuptech/space-cloud/gateway/utils"
+)
+
+// Create inserts a document (or multiple when op is "all") into the database
+func (b *Bolt) Create(ctx context.Context, project, col string, req *model.CreateRequest) (int64, error) {
+	// Create a collection object
+	// todo check if exists
+	switch req.Operation {
+	case utils.One:
+		// Insert single document
+		if err := b.client.Update(func(tx *bolt.Tx) error {
+			count, _, err := b.Read(ctx, project, col, &model.ReadRequest{
```

Comment:
If you are using the `count` operation. The `count` variable will always be one, even if no documents were present. You having to probe the result in this case.

Reply:
i don't think so,
my implementatin 
![image](https://user-images.githubusercontent.com/24411676/72268486-e9b8c600-3647-11ea-9b8b-8731c8f29ce1.png)

Reply:
You've implement count incorrectly. Check the implementation for other databases